app/models/respondent.rb

```diff
@@ -12,9 +12,15 @@
 class Respondent < ActiveRecord::Base
   has_many :survey_responses
   has_many :responses
+  has_many :survey_execution_states
 
   # Normalizes the attribute itself before validation
   phony_normalize :phone_number
 
   validates_plausible_phone :phone_number, presence: true
+  validates_uniqueness_of :phone_number
+
+  def accepting_surveys?
+    !survey_execution_states.where.not(status: SurveyExecutionState.statuses[:finished]).any?
```

Comment:
Seems like a double negative with `!` and `not` and `any?`? Could this be `survey_execution_states.where.not(status: SurveyExecutionState.statuses[:finished]).empty?`